### DISCUSSION TITLE
module 'ffmpeg' has no attribute 'Error'

### DISCUSSION THREAD

**Discussion by KevinYie** [openai/whisper#514] [General]
Hello,

We are receiving this error when running the model.transcribe function on our environment in an HPC cluster. Previous suggestions to uninstall and reinstall ffmpeg-python have not resolved the issue unfortunately. 

Do you have any suggestions? Perhaps there is a version mismatch (we are currently using ffmpeg-python 0.2.0)

**Comment by jongwook** [comment]
Have you tried this suggestion: https://github.com/openai/whisper/discussions/487#discussioncomment-4090095?

**Reply by KevinYie** [reply]
Thank you for your response! We've tried this fix already to no success

**Reply by KevinYie** [reply]
Hello,

I just wanted to follow up on this ticket as the issue is still persisting even after applying the fix suggested.

**Reply by khalid4294** [reply]
facing the same issue, were you able to fix it?

**Reply by Pnaci** [reply]
I have the same issue, have you solved it by any chance?

**Comment by glangford** [comment]
From this thread
https://github.com/kkroening/ffmpeg-python/issues/174#issuecomment-561546739

another suggestion is 
> Make sure your Python file isn't named ffmpeg.py

in case that is relevant.

**Comment by khalid4294** [comment]
I found a way to fix it.
first of all, it turned out the issue was only occurring on vscode, when I used Jupyter Notebook, the issue disappeared. 

In order to fix the issue on VScode, I did the following:
- `pip3 uninstall ffmpeg`
- `pip3 uninstall ffmpeg-python`
- `pip uninstall ffmpeg`
- `pip uninstall ffmpeg-python`
- `brew uninstall ffmpeg`

Then:
- `pip3 install ffmpeg-python`
- `brew install ffmpeg`

the issue disappeared after this, it turned out I installed ffmpeg through pip3, pip and brew.
hope it fixes your issue as well!

**Reply by walulyafrancis** [reply]
This worked for me on ubuntu 22 and python 3.8, thank you kev

**Reply by aster-fung** [reply]
This worked for me on windows 10 conda environment too thanks.

**Reply by ytang5** [reply]
Worked for me on Mac. pip install ffmpeg-python（instead of ffmpeg) solve my problem

**Reply by arieefrachman** [reply]
Awesome, its worked!!, thanks

**Reply by yacaikk** [reply]
So do i，on the whindows10!

**Reply by laminda** [reply]
PyCharm is work this

**Comment by FurkanGozukara** [comment]
I have a video that explains how to use whisper it may help. i install on windows from scratch and run it

https://www.youtube.com/watch?v=msj3wuYf3d8

**Comment by KevinYie** [comment]
Thank you for the suggestions! 

Unfortunately, I've tried uninstalling and reinstalling multiple times but the same issue occurs.

**Reply by FurkanGozukara** [reply]
Set your environment path. I have video regarding that in my channel : https://www.youtube.com/@SECourses

**Reply by KevinYie** [reply]
Unfortunately, I already set the environment path to contain the path to the bin directories for both the conda environment and the ffmpeg binary. I've also updated PYTHONPATH to reflect the actual location of the site-packages for the conda environment. 

The issue is still occuring

**Reply by jongwook** [reply]
It seems the HPC environment has a different `ffmpeg` package pre-installed. To confirm,

```python
import ffmpeg
print(ffmpeg.__file__)
```

will show the `ffmpeg` module loaded by Python. To make it load the module from `ffmpeg-python`, the path that it's installed should come before the path printed from the above command, in your `PYTHONPATH`.

Alternatively, you could try bypassing the audio reading part by manually loading the audio. Depending on the format, you could also try [`wave`](https://docs.python.org/3/library/wave.html), [`PySoundFile`](https://pysoundfile.readthedocs.io/en/latest/), [`audioread`](https://github.com/beetbox/audioread), or [`librosa.load`](https://librosa.org/doc/main/generated/librosa.load.html). Since you have a numpy array of **16kHz mono** audio samples, you can pass them to `whisper.transcribe()` function instead of the file path.

**Reply by Pnaci** [reply]
I tried also everything mentioned here and nothing works. I get the same error. Have you solved it, @KevinYie ?

**Comment by tthmic** [comment]
I'm not sure if this issue has been solved but ffmpeg-python is a wrapper for the underlying package, which has to be installed separately (https://github.com/kkroening/ffmpeg-python#installing-ffmpeg). 

The following links helped me solve this issue. I'm using Windows but I'm sure this is applicable to other OS:
https://www.wikihow.com/Install-FFmpeg-on-Windows

Hope this helps!

**Comment by mustafagunel** [comment]
I got same issue and had solve like that. Uninstall and delete all ffmpeg packages in your python library.  Download [https://github.com/kkroening/ffmpeg-python] package and move ffmpeg folder in your python library package.  On my computer its path; C:\Users\{username}\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages .

**Comment by gullalc** [comment]
For me, the issue is resolved when ffmpeg is installed with conda. No other solution in this thread worked for me.

pip install ffmpeg-python
conda install -c conda-forge ffmpeg

**Reply by BJKriptal** [reply]
This finally worked for me in the terminal; here, conda is the key point.

**Reply by SsNiPeR1** [reply]
isn't there a way to do that without conda? if there is none, it's a deal breaker :(

**Reply by Ayaaaka-zheng** [reply]
thank you! The conda command works for me.
